chopper.py

```diff
@@ -119,63 +132,86 @@ def parse_args() -> dict[str, Any]:
         default=0,
     )
 
-    args = vars(parser.parse_args())
+    namespace = ChopperNamespace()
+    config = parser.parse_args(namespace=namespace)
 
-    if not args["input_path"].exists():
-        parser.error(
-            "Invalid input_path. Please specify an existing file or directory."
-        )
+    for path in config.input_paths:
+        if not path.exists():
+            parser.error(
+                "Invalid input_paths. Please specify existing file(s) and/or directory (directories)."
+            )
 
-    if not args["output_directory"].exists():
-        os.makedirs(args["output_directory"], exist_ok=True)
+    if not config.output_directory.exists():
+        config.output_directory.mkdir(exist_ok=True)
 
-    actions = ["columns", "rows", "equal", "shuffles"]
-    if not args.keys() & actions:
+    actions = {"columns", "rows", "equal", "shuffles"}
+    if len(set(config.__dict__.keys()).intersection(actions)) == 0:
         parser.error(
             "You must specify at least one action (columns, rows, equal, shuffles)."
         )
 
-    return args
+    return config
+
+
+def get_offsets(file: BufferedReader) -> np.ndarray:
+    pos = 0
+    row_offsets = []
 
+    while row := file.readline():
+        pos += len(row)
+        row_offsets.append(pos)
 
-def shuffle_file(filepath: Path, shuffles: int, config: Config) -> list[str]:
+    return np.asarray(row_offsets)
```

Comment:
After some testing on very large files, it became clear that the standard library's `random.shuffle()` was very slow. The conversion to a numpy array takes noticeably longer, but well worth it for large files given the big improvement in shuffle time.